MusicNotationKit/MusicNotationKit/Measure.swift

```diff
@@ -46,12 +46,36 @@ public struct Measure: ImmutableMeasure, Equatable {
         noteCount = immutableMeasure.noteCount
         recomputeNoteCollectionIndexes()
     }
-
+	
+	public func note(at index: Int, inSet setIndex: Int) throws -> Note {
+		let noteCollectionIndex = try noteCollectionIndexFromNoteIndex(index, inSet: setIndex)
+		var noteIndex = 0
+		if noteCollectionIndex.tupletIndex != nil {
+			noteIndex = noteCollectionIndex.tupletIndex!
+		}
+		return try notes[setIndex][noteCollectionIndex.noteIndex].note(at: noteIndex)
+	}
+
+	public mutating func replaceNote(at index: Int, inSet setIndex: Int, with note: Note) throws {
```

Comment:
Again, set the default value of setIndex to 0. Make sure that it will work fine if you have this as the second parameter like you have it. Maybe put the `setIndex` parameter at the end.